api/commands/format.ts

```diff
@@ -117,16 +185,32 @@ function formatJavaScript(text: string) {
   return jsBeautify.js(text, JS_BEAUTIFY_OPTIONS);
 }
 
+function formatSql(text: string) {
+  let formatted = sqlFormatter.format(text) as string;
+  // Unfortunately sql-formatter does not always produce final formatted output (even on plain SQL) in a single pass.
+  for (let attempts = 0; attempts < MAX_SQL_FORMAT_ATTEMPTS; attempts++) {
+    const newFormatted = sqlFormatter.format(formatted) as string;
+    if (newFormatted === formatted) {
+      return newFormatted;
+    }
+    formatted = newFormatted;
+  }
+  throw new Error(
```

Comment:
I'd argue that it would be more annoying for the formatter to throw here than for it to flip-flop, wdyt?

Reply:
If we let it flip-flop, I need to remove the error checking at the top level that the whole formatter doesn't flip-flop-flap-etc; I don't yet feel comfortable doing that because so far it has only shown up buggy formatting (where it actually breaks the code).